src/cpuid/src/template/intel/t2cl.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use arch::x86_64::msr::{ArchCapaMSRFlags, MSR_IA32_ARCH_CAPABILITIES};
+use kvm_bindings::{kvm_cpuid_entry2, kvm_msr_entry, CpuId};
+
+use crate::cpu_leaf::*;
+use crate::template::intel::validate_vendor_id;
+use crate::transformer::*;
+
+/// Sets up the CPUID entries for a given VCPU following the T2CL template.
+struct T2CLCpuidTransformer {}
+
+impl CpuidTransformer for T2CLCpuidTransformer {
+    fn entry_transformer_fn(&self, entry: &mut kvm_cpuid_entry2) -> Option<EntryTransformerFn> {
+        match entry.function {
+            leaf_0x1::LEAF_NUM => Some(crate::t2::update_feature_info_entry),
+            leaf_0x7::LEAF_NUM => Some(crate::t2::update_structured_extended_entry),
+            leaf_0xd::LEAF_NUM => Some(crate::t2::update_xsave_features_entry),
+            leaf_0x80000001::LEAF_NUM => Some(crate::t2::update_extended_feature_info_entry),
+            leaf_0x80000008::LEAF_NUM => Some(crate::t2::update_extended_feature_extensions_entry),
+            _ => None,
+        }
+    }
+}
+
+/// Sets up the CPUID entries for a given VCPU following the T2CL template.
+pub fn set_cpuid_entries(kvm_cpuid: &mut CpuId, vm_spec: &VmSpec) -> Result<(), Error> {
+    validate_vendor_id()?;
+    T2CLCpuidTransformer {}.process_cpuid(kvm_cpuid, vm_spec)
```

Comment:
With https://github.com/firecracker-microvm/firecracker/pull/3324/files#r1067218200 this would be
```suggestion
    T2CLCpuidTransformer.process_cpuid(kvm_cpuid, vm_spec)
```